go/arrow/ipc/message.go

```diff
@@ -224,12 +231,14 @@ func (r *messageReader) Message() (*Message, error) {
 	meta := flatbuf.GetRootAsMessage(buf, 0)
 	bodyLen := meta.BodyLength()
 
-	buf = make([]byte, bodyLen)
-	_, err = io.ReadFull(r.r, buf)
+	body := memory.NewResizableBuffer(r.mem)
+	defer body.Release()
```

Comment:
the message in `newMessageFromFB` calls retain on the body buffer?

Reply:
`newMessageFromFB` calls `Retain` on the body.